src/librustc_typeck/diagnostics.rs

```diff
@@ -170,6 +170,38 @@ Since `return;` is just like `return ();`, there is a mismatch between the
 function's return type and the value being returned.
 "##,
 
+E0070: r##"
+The left-hand side of an assignment operator must be an lvalue expression. An
+lvalue expression represents a memory location and includes item paths (ie,
+namespaced variables), dereferences, indexing expressions, and field references.
+
+More details can be found here:
+https://doc.rust-lang.org/reference.html#lvalues,-rvalues-and-temporaries
+
+Now, we can go further. Here are some bad examples:
+```
+const SOME_CONST : i32 = 12;
+
+fn some_function() {
+    SOME_CONST = 14; // error !
+    1=3; // error 
+}
+```
+
+And now let's give good examples:
```

Comment:
I think some explanation about why using a `static mut` instead of a `const` would be helpful here. Perhaps this can be something like "The assignment to a `const` above is incorrect because a `const` is a *constant value* and is not associated with a memory location. You may want a `static mut` instead, which looks like this:"